core/src/main/java/aima/core/environment/support/ProblemFactory.java

```diff
@@ -51,6 +48,21 @@
 				Map2DFunctionFactory.getStepCostFunction(simplifidRoadMapOfPartOfRomania));
 	}
 
+	public static Problem<GoAction, InState> getBidirectionalSearch(String initialState, final String goalLocations) {

```

Comment:
naming: this method does not return a BiDirectionalSearch as the name suggests but a BasicProblem